src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -830,50 +834,63 @@ public LdapUserDetails updateUserDetails(LdapUserDetails d) {
 
     @Override
     public GroupDetails loadGroupByGroupname(String groupname) throws UsernameNotFoundException, DataAccessException {
+        return loadGroupByGroupname(groupname, false);
+    }
+
+    @Override
+    public GroupDetails loadGroupByGroupname(String groupname, boolean fetchMembers) throws UsernameNotFoundException, DataAccessException {
         groupname = fixGroupname(groupname);
-        Set<String> cachedGroups;
+        GroupDetailsImpl cachedGroup;
         if (cache != null) {
-            final CacheEntry<Set<String>> cached;
+            final CacheEntry<GroupDetailsImpl> cached;
             synchronized (this) {
                 cached = groupDetailsCache != null ? groupDetailsCache.get(groupname) : null;
             }
             if (cached != null && cached.isValid()) {
-                cachedGroups = cached.getValue();
+                GroupDetailsImpl cachedValue = cached.getValue();
+                if (!fetchMembers || cachedValue.getMembers() != null) {
+                    cachedGroup = cachedValue;
+                } else {
+                    cachedGroup = null;
+                }
             } else {
-                cachedGroups = null;
+                cachedGroup = null;
             }
         } else {
-            cachedGroups = null;
+            cachedGroup = null;
         }
 
         // TODO: obtain a DN instead so that we can obtain multiple attributes later
```

Comment:
`GroupDetailsImpl` now has the Group's  DN. Should I delete the TODO?

Reply:
sure